### PR TITLE
ht: re-enable the real ht_grow

### PR DESCRIPTION
**Pull request by ret2libc**
test: add unit testing for almost all hashtable functionalities

-----------
Let's see if it works :)

### CODE REVIEW

**Comment by radare** [comment]
What is a dylib doing here?

**Comment by ret2libc** [comment]
I think i've removed it before... I can't see it anymore

File: test/unit/test_hash.c [hunk 1]
```diff
@@ -15,8 +15,109 @@ bool test_ht_insert_lookup(void) {
...
+}
+
+bool test_ht_grow(void) {
+	SdbHash *ht = ht_new ();
+	char str[15], vstr[15];
+	char buf[100];
+	int i;
+
+	ht->hashfn = incr_hash;
+	for (i = 0; i < 20000; ++i) {
```
Comment (radare) [inline test/unit/test_hash.c]: Lot of missing spaces before (

**Comment by ret2libc** [comment]
@radare it should be ok now

File: .gitignore [hunk 2]
```diff
@@ -13,6 +15,24 @@ test/reset/test-reset
...
+test/merge
+test/milset
+test/milsetv
+test/ns
+test/nsabuse
+test/siolpain
+test/syncget
+test/test.db
+test/unit/test_hash
+test/unit/test_hash.dSYM/
```
Comment (radare) [inline .gitignore]: why not just ignore all *.dSYM

**Pull request merged**